modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -672,6 +676,39 @@ public void testDeleteUsingCompositePk() {
         assertQuery("SELECT d FROM test").returnNothing().check();
     }
 
+    @Test
+    public void testOperationFailsToCommitWhenDetectsSchemaChange() throws InterruptedException {
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+
+        CountDownLatch startDml = new CountDownLatch(1);
+        CountDownLatch latch = new CountDownLatch(1);
+
+        CompletableFuture<Void> upsertFut = CompletableFuture.runAsync(() -> {
```

Comment:
I ran this test 10 times on the main branch  without failures... so I'm not sure if multithreading is working properly here.

Reply:
> somehow modify the test so that it reproduces the original error,
it seems impossible to make a stable test for the original error :pensive:  so we can probably leave the current implementation in the hope that it will catch something someday.

Reply:
Removed this test.